Test Selection Page HTML and CSS layout

chrisn02 opened the pull request:
Have created the basic layout for the test selection page. Each test box changes colour when hovered over and when clicked you are taken to the testing page. Potential for more design to be done, but have general page layout. Any feedback or changes appreciated
![Test Selection Page Layout](https://github.com/Jforjo/CSD/assets/59711723/7b0f10e8-1b76-42ff-be22-5abbae16f8f8)
![Test Selection Page Layout Highlighted](https://github.com/Jforjo/CSD/assets/59711723/07acc07c-0f77-4079-953f-344ae58968c5)

chrisn02 committed:
Dashboard page started

chrisn02 committed:
Added "Tests to complete" & "Completed Tests" section

chrisn02 committed:
Added some styling to test boxes

Made changes such as adding a border around each test and making it so each test is next to each other rather than underneath.

chrisn02 committed:
Made test box slightly bigger

chrisn02 committed:
Change file extension to HTML (temp)

chrisn02 committed:
Made it so that the tests are centred and when you click it, you are taken to the testing page

Also moved styling to its own css file and renamed file back to PHP.

chrisn02 committed:
Done some more styling

Added footer section as well as some general styling changes

Jforjo commented on style.css:
```diff
@@ -0,0 +1,60 @@
+*
```
Just some advice: you might want to have `margin: 0`, `padding: 0` and `box-sizing: border-box` here.

gchodancar19 replied:
Looks good to me.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,60 @@
+    {
+        font-family: Arial, Helvetica, sans-serif;
+        color: white;
+        background-color: #181818;
```
You should not have a `background-color` on everything. Just the body is fine.

Jforjo commented on dashboard.php:
```diff
@@ -0,0 +1,84 @@
+                <p>Score: 80%</p>
+                <p>250 points</p>
+            </div>
+            </a>
```
This closing `</a>` tag should not be here.

Jforjo commented on dashboard.php:
```diff
@@ -0,0 +1,84 @@
+    <section>
+        <h2 class="section-title">Completed Tests</h2>
+            <div class="test-box">
+            <a href="testing-page.php" class="test-link">
```
This line should have a closing `</a>` tag.

Jforjo commented on dashboard.php:
```diff
@@ -0,0 +1,84 @@
+            </div>
+            </a>
+            </div>
+        </a>
```
This closing `</a>` tag should not be here.

Jforjo commented on dashboard.php:
```diff
@@ -0,0 +1,84 @@
+    <section>
+        <h2 class="section-title">Tests to complete</h2>
+            <div class="test-box">
+            <a href="testing-page.php" class="test-link">
```
This line should have a closing `</a>` tag.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,60 @@
+    {
+        margin: 0 20px;
+    }
+
```
I think it would be a nice feature to remove the underline on the navbar links using `text-decoration: none`, but you could show them again on `:hover` with `text-decoration: underline`.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,60 @@
+        text-decoration: none;
+    }
+
+    .test-box a
```
It might be good to centre your content on the page, as right now it's not.

Jforjo commented on dashboard.php:
```diff
@@ -0,0 +1,84 @@
+        </a>
+        
+    </section>
+    <section>
```
I think the tests that the user has already completed should be hidden so they only see the tests they have left to do, but the completed tests could be revealed when a button is pushed.

gchodancar19 replied:
I think the same.

Jforjo reviewed (changes requested):
Overall, good template, but does need some fundamental changes. See the comments on each line (some of them are just suggestions).
I also think you could include some colour on the website, or just more 'stuff' - it feels too minimalistic.
Finally, the CSS could do with a lot of attention, but this looks to be a base design for now, so consider it in the future.

chrisn02 commented:
Ok thank you for your feedback, I will make the changes mentioned. Any suggestions to make it less minimalistic?

Jforjo commented:
For starters, I'd recommend just browsing [Dribbble.com](https://dribbble.com) for some inspiration. (Yes, there are 3 b's in Dribbble)
You could add some more shades of black to show elevation, like the test cards would have a higher elevation to the header/navigation bar (the lighter the shade, the higher the elevation. Just don't go too high too quickly)
You might also want to add some colour, usually a bright colour that stands out, as this will be your call-to-action colour. Most probably on your buttons that might say "Start Test" or something. This will draw the users' attention to it as something they need to interact with.

chrisn02 commented:
Thanks, will take a look.

chrisn02 committed:
Design changes and added "Start Test" button

Making some general changes that were requested as per pull request ID #3

chrisn02 committed:
Made changes to test box

Changed what happens when hovering over the test box from changing colour to getting slightly bigger. Also spaced out the subject name and test name sections within the box

chrisn02 committed:
Added welcome section and centred contents

chrisn02 committed:
Separated test section into own class

Separated the test box section into its own class rather than having the test box itself unintentionally being used for the entire section

chrisn02 committed:
More design changes

Made it so that the colour from the navbar is slightly lighter and it blends into the colour of the main page. Also made some changes to the test box such as made it so the Test Subject is now at the top and is separated by an underline

chrisn02 committed:
Test boxes change size based off screen size

Also some general changes to text box itself such as smoothing out the corners a bit

chrisn02 committed:
Changed layout of contents in test box

Might redo at some point but for now this is the general layout

chrisn02 commented:
Have made some design changes to the page. Added an effect where the colour from the navbar fades into the colour of the main page. Also added a welcome section and revamped the test boxes, there is now a start test button that the user must click on to start the test rather than clicking on the entire test box. There is also a slight shadow to the boxes and when hovered over, it increases in size slightly and a glow effect appears. The actual contents of the test box could still do with a few changes but can be done at a later date. Let me know your thoughts.
![TestSelectionPage_Updated](https://github.com/Jforjo/CSD/assets/59711723/3c155f56-b828-40e9-a224-f68b2291314b)
![TestSelectionPage_Updated_Hover](https://github.com/Jforjo/CSD/assets/59711723/af9978b4-0520-4a96-8c18-77180cc074fb)

Jforjo commented:
At the top of the page on the left next to my name there should be a "re-request review" button, click that if you feel like this pull request is completed and you want to push it to sudo-main.

Jforjo reviewed (approved):
Design improved, but as you said it can be improved more - you might get inspiration when you start working on the backend and realise there could be a better way to display something.
You have only used "btn btn-primary" from Bootstrap, so instead of loading the entire Bootstrap library you could just style the buttons yourself. This would increase page load times, albeit not by much.

chrisn02 commented:
Thanks. Will definitely try to improve on it when it comes to working on backend and will also look at switching the button from bootstrap to own styling

gchodancar19 reviewed (approved):
It looks good to me.
